src/System.Net.Http/src/System/Net/Http/HttpClientHandler.Unix.cs

```diff
@@ -160,6 +161,10 @@ protected override void Dispose(bool disposing)
 
         protected internal override Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
         {
+            if (DiagnosticsHandler.IsEnabled())
```

Comment:
This ends up calling:
```
s_diagnosticListener.IsEnabled(DiagnosticsHandlerLoggingStrings.RequestWriteName); 
```
How much slower is that than just:
```
s_diagnosticListener.IsEnabled();
```
Basically, I'm looking for the fastest possible way to know "diagnostics are definitely not enabled": it's ok if we report that diagnostics are enabled even if this particular listener may not be, but it's not ok to report that diagnostics are not enabled even if they may be.  That then ensures that if no diagnostics are enabled, we as quickly as possible can take the fast path of going directly to CurlHandler (or WinHttpHandler), but if any diagnostics are enabled, then we can take the slightly slower (hopefully not too much slower) path of going through the extra checks.